src/dscigametrics/metric4.py

```diff
@@ -0,0 +1,26 @@
+def find_best_and_worst_campaigns(df, start_date, end_date, campaign_ids, metric):
+    '''
+
+    Analyzes and identifies the best and worst performing marketing campaigns based on a selected metric from Google Analytics data over a specified date range.
+
```

Comment:
Also add the four metrics description in here, you can copy from Ian's script (compute_metrics)